models/balancer/arbitrum/balancer_arbitrum_pools_tokens_weights.sql

```diff
@@ -0,0 +1,37 @@
+{{
+    config(
+        alias='pools_tokens_weights',
+        post_hook='{{ expose_spells(\'["arbitrum"]\',
+                                    "project",
+                                    "balancer",
+                                    \'["metacrypto", "jacektrocinski"]\') }}'
+    ) 
+}}
+
+--
+-- Balancer v2 Pools Tokens Weights
+--
+SELECT
+    registered.`poolId` AS pool_id,
+    tokens.token_address,
+    weights.normalized_weight / POWER(10, 18) AS normalized_weight
+FROM {{ source('balancer_v2_arbitrum', 'Vault_evt_PoolRegistered') }} registered
+INNER JOIN {{ source('balancer_v2_arbitrum', 'WeightedPoolFactory_call_create') }} call_create
+    ON call_create.call_tx_hash = registered.evt_tx_hash
+    LATERAL VIEW posexplode(call_create.tokens) tokens AS pos, token_address
+    LATERAL VIEW posexplode(call_create.weights) weights AS pos, normalized_weight
+WHERE tokens.pos = weights.pos
+
+UNION ALL
+
+SELECT
+    registered.`poolId` AS pool_id,
+    tokens.token_address,
+    weights.normalized_weight / POWER(10, 18) AS normalized_weight
+FROM {{ source('balancer_v2_arbitrum', 'Vault_evt_PoolRegistered') }} registered
+INNER JOIN {{ source('balancer_v2_arbitrum', 'WeightedPool2TokensFactory_call_create') }} call_create
+    ON call_create.call_tx_hash = registered.evt_tx_hash
+    LATERAL VIEW posexplode(call_create.tokens) tokens AS pos, token_address
+    LATERAL VIEW posexplode(call_create.weights) weights AS pos, normalized_weight
+WHERE tokens.pos = weights.pos
+;
```

Comment:
Let's simplify this query using a `for` loop as we do on `dex_trades`. Also, we didn't update this view on V1 with the new `WeightedPoolV2Factory` -- could you add it here? 

Remember you'll have to do this on Optimism and Polygon as well.

Reply:
I didn't use a `for` loop since the `weights` column has a different name in `WeightedPoolV2Factory_call_create`.